src/Time/TakeStep.hpp

```diff
@@ -0,0 +1,27 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#pragma once
+
+#include "DataStructures/DataBox/DataBox.hpp"
+#include "Parallel/GlobalCache.hpp"
+#include "Time/Actions/ChangeStepSize.hpp"
+#include "Time/Actions/RecordTimeStepperData.hpp"
+#include "Time/Actions/UpdateU.hpp"
+#include "Utilities/Gsl.hpp"
+
+/// Bundled method for recording the current system state in the history, and
+/// updating the evolved variables and step size.
+///
+/// This function is used to encapsulate any needed logic for updating the
+/// system, and in the case for which step parameters may need to be rejected
+/// and re-tried, looping until an acceptable step is performed.
+template <typename StepChooserRegistrars, typename VariablesTag = NoSuchType,
+          typename DbTags, typename Metavariables>
+void take_step(const gsl::not_null<db::DataBox<DbTags>*> box,
+               const Parallel::GlobalCache<Metavariables>& cache,
+               const VariablesTag /*meta*/ = VariablesTag{}) noexcept {
```

Comment:
Why do you need this as an argument?  Doesn't the template parameter work on its own?

Reply:
You're right -- it's a better interface to just always specify the template argument, I'll omit the extra argument.